pkg/controller/control.go

```diff
@@ -1251,16 +1251,17 @@ func (c *Controller) handleDiskNoSpaceErrorForReplicas(replicaNoSpaceErrMap map[
 
 	replicaWritableCount := 0
 	for _, r := range c.replicas {
-		if r.Mode == types.ERR {
+		if r.Mode != types.RW {
 			continue
 		}
 		replicaWritableCount++
 	}
 
 	// https://github.com/longhorn/longhorn/issues/10718#issuecomment-3190666485
 	// If a replica is on a disk that is out of space, and others are not, this replica will be marked as ERR to avoid the inconsistent data.
-	// The volume will be readable when all disks of the replicas are out of space so last one replica will not be marked as ERR because of ENOSPC.
-	if len(replicaNoSpaceErrMap) == replicaWritableCount {
+	// If all replicas are on disks that are out of space at the same time, the volume will be directly return an ENOSPC error and no replica will be set to ERR.
+	// Last one replica will not be marked as ERR for an ENOSPC error so the volume will still be readable when all disks of the replicas are out of space.
+	if len(replicaNoSpaceErrMap) == replicaWritableCount || replicaWritableCount == 1 {
 		return types.ErrNoSpaceLeftOnDevice
```

Comment:
It might help to return a different error when `replicaWritableCount == 1`, so it’s easier to spot the single-replica case in the higher-level logs.

Reply:
Agree. @mantissahz Can you add a log message here?